Open Judge System/OJS.Services/Data/Participants/ParticipantsDataService.cs

```diff
@@ -22,56 +22,60 @@ public class ParticipantsDataService : IParticipantsDataService
         public Participant GetById(int id) => this.participants.GetById(id);
 
         public Participant GetByContestByUserAndByIsOfficial(int contestId, string userId, bool isOfficial) =>
-            this.GetAll()
-                .FirstOrDefault(p =>
-                    p.ContestId == contestId &&
-                    p.UserId == userId &&
-                    p.IsOfficial == isOfficial);
+            this.GetAllByContestByUserAndIsOfficial(contestId, userId, isOfficial)
+                .FirstOrDefault();
 
         public Participant GetWithContestByContestByUserAndIsOfficial(int contestId, string userId, bool isOfficial) =>
-            this.GetAll()
+            this.GetAllByContestByUserAndIsOfficial(contestId, userId, isOfficial)
                 .Include(p => p.Contest)
-                .FirstOrDefault(p =>
-                    p.ContestId == contestId &&
-                    p.UserId == userId &&
-                    p.IsOfficial == isOfficial);
+                .FirstOrDefault();
 
         public IQueryable<Participant> GetAll() => this.participants.All();
 
+        public IQueryable<Participant> GetAllByUser(string userId) =>
+            this.GetAll()
+                .Where(p => p.UserId == userId);
+
+        public IQueryable<Participant> GetAllByContest(int contestId) =>
+            this.GetAll()
+                .Where(p => p.ContestId == contestId);
+
         public IQueryable<Participant> GetByIdQuery(int id) =>
             this.GetAll()
                 .Where(p => p.Id == id);
 
+        public IQueryable<Participant> GetAllOfficialByContest(int contestId) =>
+            this.GetAllByContest(contestId)
+                .Where(p => p.IsOfficial);
+
         public IQueryable<Participant> GetAllOfficialInOnlineContestByContestAndContestStartTimeRange(
             int contestId,
             DateTime contestStartTimeRangeStart,
             DateTime contestStartTimeRangeEnd) =>
-            this.GetAll()
+            this.GetAllOfficialByContest(contestId)
```

Comment:
Not sure what the established convention is, however I think this should be indented further, to visually separate the body from the arguments.